caravel/views.py

```diff
@@ -1833,6 +1841,32 @@ def csv(self, client_id):
         return response
 
     @has_access
+    @expose("/fetch_datasource_metadata")
+    @log_this
+    def fetch_datasource_metadata(self):
+        # TODO: check permissions
```

Comment:
I think it's only checking access to the endpoint, but we need access checking for datasource/slice here